components/item-info.js

```diff
@@ -33,10 +35,19 @@ export default function ItemInfo ({
   const [canEdit, setCanEdit] =
     useState(item.mine && (Date.now() < editThreshold))
   const [hasNewComments, setHasNewComments] = useState(false)
+  const [isParentCrossposted, setIsParentCrossposted] = useState(false)
   const [meTotalSats, setMeTotalSats] = useState(0)
   const root = useRoot()
   const sub = item?.sub || root?.sub
 
+  const { data } = useQuery(ITEM_FULL, { variables: { id: item?.parentId } })
```

Comment:
The performance implications of this are significant. For every comment, it queries the server for the entire parent. In a comment thread with 100 comments, this will make 100 requests to the server.

I think what we want to do instead is when rendering a `Comment` component from a parent item/comment, send it this info as props.

When a comment is in list of comments like on a profile, you'll want to use a different query fragment with a nested resolver for `parent` so that this data is gathered when all the other data is. An example of this is the nested resolver for `root`.

Reply:
Sweet thanks for the feedback! I'll get right on this